### PR TITLE
Create "PhpLabel" rule

### LINKED ISSUE #401: New rule: String that is a valid PHP label (variable/function names, etc.)
Would be good to be able to validate that a string is a valid PHP label - useful when writing something that will take input to help create output that we expect to execute later.

From the PHP docs:

http://php.net/manual/en/functions.user-defined.php
http://www.php.net/manual/en/language.variables.basics.php

````
As a regular expression, it would be expressed thus: [a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*
````

### PR DESCRIPTION
**Pull request by batusa**
have a look at my commit message.

is anything else missing?

***
Closes #403 and fix #401.

### CODE REVIEW

File: docs/PhpLabel.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# PHP Label
```
Comment (henriquemoody) [inline docs/PhpLabel.md]: _PhpLabel_ just to follow the same standard we have on other documents.

File: tests/integration/phplabel.phpt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+--TEST--
+PHP label rule
+--FILE--
+<?php
+require 'vendor/autoload.php';
+
+use Respect\Validation\Validator as v;
+
+var_dump(v::phpLabel()->validate('person'));
+var_dump(v::phpLabel()->validate('4ccess'));
```
Comment (henriquemoody) [inline tests/integration/phplabel.phpt]: This is already covered by the unit tests, would be nicer to test check and/or assert methods.

Comment (batusa) [inline tests/integration/phplabel.phpt]: I'll do it as soon as I can.

thanks!!

**Comment by henriquemoody** [comment]
Hi, @batusa! 

You asked if is anything missing, this pull request is good already, but I will not merge it until if fulfill my checklist:

- [x] Create rule class
- [x] Create rule exception class
- [x] Create a test case class for the rule
- [x] Create a markdown document with the documentation of the rule
- [ ] Add rule to the `Validator`'s class docblock
- [ ] Add _See also:_ section to the rule documentation
- [ ] On "docs/VALIDATORS.md" add link to the rule documentation in the proper section (s)
- [ ] On "docs/VALIDATORS.md" add link to the rule documentation in the "Alphabeticaly" section
- [ ] Use `RuleTestCase` to test rule instead of `PHPUnit_Framework_TestCase`
- [ ] Create integration for when exception should not be thrown
- [ ] Create integration for when exception should be thrown by `check()` method
- [ ] Create integration for when exception should be thrown by `assert()` method
- [ ] Create integration with `not()` rule for when exception should be thrown by `check()` method
- [ ] Create integration with `not()` rule for when exception should be thrown by `assert()` method

This is too much to ask for our contributors, that's why I usually do it myself like on #573.

If you want to make it 100% _mergeable_ I can wait for the changes or I can do it myself. It's up to you and I will be glad on both of them. :smile:

**Comment by batusa** [comment]
no problem, I can do it, tonight! ;)

**Comment by batusa** [comment]
"Add See also: section to the rule documentation"

what rules do you recommend to put in this part?

**Comment by batusa** [comment]
@henriquemoody please, see if I did everything right. Thanks!

**Comment by henriquemoody** [comment]
Awesome, @batusa! Thanks a lot for contributing! :panda_face: 

I merged it manually by f40eb63d7b640697f612d513ea47b61c6a3f6293. 

I just squashed the commits at one and ran php-cs-fixer.

**Comment by batusa** [comment]
o/

**Pull request closed without merging**